### PR TITLE
Cache the JSON string of BridgeChannelMessage.

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt [hunk 1]
```diff
@@ -188,6 +202,10 @@ class ServerHelloMessage @JvmOverloads constructor(
 class EndpointMessage(val to: String) : BridgeChannelMessage(TYPE) {
     @JsonInclude(JsonInclude.Include.NON_NULL)
     var from: String? = null
+        set(value) {
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:205]: The annotation can not be applied to a `var` with a delegate

File: jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt [hunk 2]
```diff
@@ -241,10 +259,12 @@ class ReceiverVideoConstraintMessage(val maxFrameHeight: Int) : BridgeChannelMes
 /**
  * A message sent from the bridge to a client, indicating that the dominant speaker in the conference changed.
  */
+@JsonInclude(JsonInclude.Include.NON_NULL)
 class DominantSpeakerMessage @JvmOverloads constructor(
-    var dominantSpeakerEndpoint: String,
+    dominantSpeakerEndpoint: String,
     val previousSpeakers: List<String>? = null
 ) : BridgeChannelMessage(TYPE) {
+    var dominantSpeakerEndpoint: String by observableWhenChanged(dominantSpeakerEndpoint) { -> resetJsonCache() }
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:267]: Apparently this is a consice way to diambiguate between `_ -> ...`  and `_, _, _ -> ...`

File: jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt [hunk 3]
```diff
@@ -241,10 +259,12 @@ class ReceiverVideoConstraintMessage(val maxFrameHeight: Int) : BridgeChannelMes
 /**
  * A message sent from the bridge to a client, indicating that the dominant speaker in the conference changed.
  */
+@JsonInclude(JsonInclude.Include.NON_NULL)
 class DominantSpeakerMessage @JvmOverloads constructor(
-    var dominantSpeakerEndpoint: String,
+    dominantSpeakerEndpoint: String,
```
Comment (JonathanLennox) [inline jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:264]: Does `dominantSpeakerEndpoint` actually need to be a `var`?  I don't see it being set anywhere, other than `BridgeChannelMessageTest`.

Comment (bgrozev) [reply jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:264]: It's just the test. I can remove it

**Comment by bbaldino** [comment]
What are the multiple places that are serializing the message?

**Comment by JonathanLennox** [comment]
> What are the multiple places that are serializing the message?

```
        for (AbstractEndpoint endpoint : endpoints)
        {
                endpoint.sendMessage(msg);
        }
```

**Comment by bgrozev** [comment]
> What are the multiple places that are serializing the message?

That's client stats being broadcast indirectly, and dominant speaker notification broadcast via `broadcastMessage`

**Comment by bbaldino** [comment]
> 
> ```
>         for (AbstractEndpoint endpoint : endpoints)
>         {
>             endpoint.sendMessage(msg);
>         }
> ```

> That's client stats being broadcast indirectly, and dominant speaker notification broadcast via broadcastMessage

Makes sense, thanks.

**Pull request merged**